Source/utils/display.cpp

```diff
@@ -361,7 +361,11 @@ void ReinitializeTexture()
 	auto quality = StrCat(static_cast<int>(*sgOptions.Graphics.scaleQuality));
 	SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, quality.c_str());
 
+#ifdef PSP
+	texture = SDLWrap::CreateTexture(renderer, SDL_PIXELFORMAT_ABGR1555, SDL_TEXTUREACCESS_STREAMING, gnScreenWidth, gnScreenHeight);

```

Comment:
A bit raw for now until I get a chance to test what works best on hardware, should move it to a define before merging.

Reply:
We now have `DEVILUTIONX_DISPLAY_TEXTURE_FORMAT` in CMake that we can use for this

Reply:
it might. psp supports 565, 5551 and 8888 afaik. I guess the problem is should we always select the lowest, highest or another? Or should we make this an option for the user in settings.
Often it really just makes sens to pick 1 for a given platform rather then having each user experiment with it.

Reply:
I think the important thing is to ensure that the requested pixel format is supported natively - the PSP (and PS2) supports `SDL_PIXELFORMAT_ABGR8888` but not `SDL_PIXELFORMAT_XRGB888`, so requesting the latter format requires any data copied to the texture to be converted in software before it can be uploaded to the GPU. There might also be bandwidth advantage to using a 16-bit format over a 32-bit one, but other than that the specifics shouldn't matter too much.

That said, since devilutionX seems to be designed around palette-based displays, libsdl-org/SDL#6192 may be of interest here. It still needs more work before it can be merged, though.

Reply:
Ok, that isn't to much of a worry though, SDL_PIXELFORMAT_ABGR8888 is simply to slow on both platforms.

Hope you manage to finish the PR, could be useful if supported on most platforms.